assets/js/googlesitekit/widgets/util/separate-null-widgets.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Widgets separation utilities.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export function separateNullWidgets( widgets, widgetStates ) {
+	return widgets.reduce( ( separated, widget ) => {
+		if ( 'WidgetNull' !== widget.Component().type.name ) {
+			// @TODO figure out why we have a null state in widgetStates when the component is not WidgetNull
+			separated.activeWidgets.push( widget );
+		} else if ( undefined === widgetStates[ widget.slug ] || null === widgetStates[ widget.slug ] ) {
+			separated.inactiveWidgets.push( widget );
+		} else {
+			separated.activeWidgets.push( widget );
+		}
```

Comment:
These comparisons should only use the `widgetState` for the widget rather than touching the `widget.Component`. Keep in mind that the component used by the widget's state for `WidgetNull` will be `Null`. We need to import that here and simply check `widgetState[ slug ]?.Component === Null` where `Null` is the actual component.

For inactive, I think we can simply check `! widgetStates[ widget.slug ]` since any state would be set as an object here.

Reply:
Removed.